midend/interpreter.cpp

```diff
@@ -877,9 +877,22 @@ void ExpressionEvaluator::postorder(const IR::Operation_Unary* expression) {
     } else if (l->is<SymbolicBool>()) {
```

Comment:
These changes look much better.  This example contains cast to bit<1>. So, the `get` [here](https://github.com/p4lang/p4c/blob/1750e2f36aa73b64c0707006f15a24cf75d25c79/midend/interpreter.cpp#L848) should return SymbolicInteger type, but not a SymbolicBool. In this case, the additional checking for a Constant will be not necessary below.